Django 2.2 support.

sasha0 opened the pull request:
Refs #3008

pjstevns commented:
:+1:

SrdjanCosicPrica commented:
Apologies for a newbie question. But doesn't `setup.py` need to be changed as well?
Right now it is 
```
install_requires = [
    'django>=1.11,<2.2',
    ...
```

maerteijn commented:
@SrdjanCosicPrica No it should be adjusted indeed as well

solarissmoke commented on src/oscar/apps/customer/utils.py:
```diff
@@ -132,9 +132,12 @@ def get_password_reset_url(user, token_generator=default_token_generator):
     Generate a password-reset URL for a given user
     """
+    uidb64 = urlsafe_base64_encode(force_bytes(user.id))
+    if isinstance(uidb64, bytes):
```
We will almost certainly forget to remove this check when we drop support for Django 2.1 and below. I think it is better to wrap `urlsafe_base64_encode` in a helper method in `oscar.core.compat`, so we know it can be removed later without leaving dead code around.

sasha0 replied:
Ok done.

sasha0 commented:
> This is however mentioned in the comment in catious_slugify and used in templates here and here. So maybe we should add |safe in the templates and update the comment?

Do we really need to add safe filter there, since slugify is not supposed to return any HTML characters?

> I also saw that you added xenial to the travis configuration: Then you could also add python 3.7 as well (that works with xenial).

Ok done.

maerteijn commented:
No I don’t think so, just mentioned it as an option. Then I would suggest to remove the mark_safe part in the comment.

maerteijn commented:
> > I also saw that you added xenial to the travis configuration: Then you could also add python 3.7 as well (that works with xenial).
> 
> Ok done.

I think you should adjust `tox.ini`:
`py{35,36,37}-django{111,20,21}` -> add `22` as well

And refer to the correct tox envs in `.travis.yml`:
```
    - python: 3.7
      env: TOXENV=py36-django22
``` 
(should be `TOXENV=py37-django22`)

sasha0 commented:
> No I don’t think so, just mentioned it as an option. Then I would suggest to remove the mark_safe part in the comment.

Right. Updated.

> I think you should adjust tox.ini:
py{35,36,37}-django{111,20,21} -> add 22 as well
>
>And refer to the correct tox envs in .travis.yml:
>
>    - python: 3.7
>      env: TOXENV=py36-django22
>(should be TOXENV=py37-django22)

Ok, adjusted.

maerteijn commented:
Looks fine to me, so @solarissmoke should review (and hopefully approve) your `urlsafe_base64_encode` implementation and then we can merge right?

sasha0 commented:
Yep.
